src/lsm/merge_iterator.zig

```diff
@@ -55,7 +56,9 @@ pub fn MergeIteratorType(
                             continue;
                         },
                     };
-                    return it.iterator_b.pop();
+                    const value_b = it.iterator_b.pop();
+                    assert(it.iterator_b_may_contain_tombstones or !Table.tombstone(&value_b));
```

Comment:
Would it work to assert these in `compaction.zig` when the merge iterator is popped, rather than within the iterator itself? That would avoid the extra parameter to `MergeIterator`.

Reply:
If there is a value in `a` and a tombstone in `b`, only the value is returned from the iterator.

Reply:
Ah right. Maybe group `MergeIterator`'s parameters into a struct? (I'm trying to avoid a boolean argument that is unlabelled on the caller side, since it doesn't have an obvious meaning w/o looking it up in the definition).

Reply:
Good point.

The formatting is unfortunate, but I couldn't figure out a better option :S